src/models/Permissions/UserPermissions/Permissions.ts

```diff
@@ -0,0 +1,17 @@
+import { Offer } from "$models";
+import { IPermission } from "../Interface";
+import { CurrentUser } from "$models/CurrentUser";
+
+export class UserPermissions implements IPermission {
+  private readonly permissions: IPermission[];
+
+  constructor(currentUser: CurrentUser) {
+    this.permissions = currentUser.roles.map(role => role.getPermissions());
+  }
+
+  public async canSeeOffer(offer: Offer) {
+    return (
+      await Promise.all(this.permissions.map(permission => permission.canSeeOffer(offer)))
+    ).some(canSeeOffer => canSeeOffer);
```

Comment:
![image](https://user-images.githubusercontent.com/20487262/93662462-9d275500-fa36-11ea-9e55-97bba4f05d9e.png)

lodash da algo mejor para esto

Reply:
aca podes delegar, tipo

`this.hasPermission(permission => permission.canSeeOffer(offer))`

`hasPermission (callback) => some(await Promise.all(this.permissions.map(callback)))`